src/main/java/racingcar/controller/MainController.java

```diff
@@ -1,30 +1,43 @@
 package racingcar.controller;
 
-import racingcar.domain.Car;
-import racingcar.domain.CarRacingGame;
-import racingcar.domain.RoundResult;
-import racingcar.dto.CarDto;
+import racingcar.domain.*;
+import racingcar.utils.Converter;
 import racingcar.view.InputView;
 import racingcar.view.OutputView;
 
 import java.util.List;
 import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
 public class MainController {
 
     public void run() {
-        CarRacingGame carRacingGame = repeat(this::initializeCarRacingGame);
-        List<RoundResult> roundResults = carRacingGame.race();
+        CarRacingGame carRacingGame = initializeCarRacingGame();
+        List<RoundResult> roundResults = carRacingGame.race(new RandomNumberGenerator());
 
         showRoundResults(roundResults);
         showWinners(carRacingGame.findWinners());
     }
 
     private CarRacingGame initializeCarRacingGame() {
+        Cars cars = repeatTemplate(this::initializeCars);
+        Round round = repeatTemplate(this::initializeRound);
+
+        return new CarRacingGame(cars, round);
+    }
+
+    private Cars initializeCars() {
         String inputCarNames = InputView.inputCarNames();
-        String inputRound = InputView.inputRound();
+        List<String> carNames = Converter.convert(inputCarNames);
+
+        return carNames.stream()
+                .map(Car::new)
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Cars::new));
```

Comment:
List<Car> 를 만들어서 생성자로 Cars를 만드는것이
더 가독성이 좋지 않을까요?

Reply:
저도 처음에 그렇게 구현하였어요!! 🧐

하지만 생성자로 Cars를 만드는 방법의 문제점은 `자동차들의 이름이 중복되었을 경우`에도, 시도할 횟수까지 입력받고 다시 이름을 입력받기 때문에 Cars를 먼저 생성하도록 구현하였어요!

Reply:
```suggestion
        List<Car> cars = carNames.stream()
            .map(Car::new)
            .toList();

        return new Cars(cars);
```
이런 의도였는데 시도할 횟수까지 입력을 받는다는게 무슨의미일까요?
new Cars에서 validate되는거 아닌가요??🤔

Reply:
아 제가 잘못 이해했네요ㅠㅠ

```java
public class Cars {
    private final List<Car> cars;

    public Cars(List<String> carNames) {
        this.cars = carNames.stream()
                .map(Car::new)
                .toList();
    }
}
```
다음과 같이 Cars 생성자 내부에서 List<Car>를 생성하라는 뜻으로 이해했었습니다...ㅠㅠ
영이가 알려준 방식대로 수정해보았어요!! 👍